junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -52,8 +52,11 @@ static void fail(String message, Object expected, Object actual) {
 		throw new AssertionFailedError(message, expected, actual);
 	}
 
-	static String nullSafeGet(Supplier<String> messageSupplier) {
-		return (messageSupplier != null ? messageSupplier.get() : null);
+	static String nullSafeGet(Object messageContainer) {
```

Comment:
Unless I'm missing something, we should retain the original `nullSafeGet(Supplier<String>)` method and simply introduce an overloaded version that accepts a `String`.

Reply:
I'm adding back `nullSafeGet(Supplier<String> messageSupplier)` and keeping the overloaded version `String nullSafeGet(Object messageContainer)`. Not a `String` as you mentioned since if I make it string I'll need to copy/paste more than 11 methods. I.e. `AssertIterableEquals#assertIteratorsAreEmpty`. Please note that `Object messageContainer` is used only in private methods. The only place it's package private is `AssertionUtils.nullSafeGet(Object messageContainer)`.

Reply:
OK. I'll have a look at this as well when I review on my machine.